src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -121,6 +121,160 @@ public Vector2 Center
             set => Translate(value - Center);
         }
 
+        // --
+
+        public float Width
+        {
+            get => _max.X - _min.X;
+            set => _max.X = _min.X + value;
+        }
+
+        public float Height
+        {
+            get => _max.Y - _min.Y;
+            set => _max.Y = _min.Y + value;
+        }
+
+        public float Left
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Top
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float Right
+        {
+            get => _max.X;
+            set => _max.X = value;
+        }
+
+        public float Bottom
+        {
+            get => _max.Y;
+            set => _max.Y = value;
+        }
+
+        public float X
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Y
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float SizeX
+        {
+            get => _max.X - _min.X;
+        }
+
+        public float SizeY
+        {
+            get => _max.Y - _min.Y;
+        }
+
+        public Vector2 Location => _min;
+
+        public bool IsEmpty => _min.X == 0 && _min.Y == 0 && _max.X == 0 && _max.Y == 0;
+
+        public static readonly Box2 Zero = new Box2(0, 0, 0, 0);
+
+        public static readonly Box2 Identity = new Box2(0, 0, 1, 1);
+
+        public static Box2 FromSize(Vector2 location, Vector2 size)
+        {
+            return new Box2(location, location + size);
+        }
+
+        public static Box2 FromPositions(Vector2 min, Vector2 max)
+        {
+            return new Box2(min, max);
+        }
+
+        public static Box2 FromPositions(float minX, float minY, float maxX, float maxY)
+        {
+            return new Box2(minX, minY, maxX, maxY);
+        }
+
+        public void Intersect(Box2 other)
+        {
+            Box2 result = Box2.Intersect(other, this);
+
+            X = result.X;
+            Y = result.Y;
+            Width = result.Width;
+            Height = result.Height;
+        }
+
+        public static Box2 Intersect(Box2 a, Box2 b)
+        {
+            float minX = a._min.X > b._min.X ? a._min.X : b._min.X;
+            float minY = a._min.Y > b._min.Y ? a._min.Y : b._min.Y;
+            float maxX = a._max.X < b._max.X ? a._max.X : b._max.X;
+            float maxY = a._max.Y < b._max.Y ? a._max.Y : b._max.Y;
+
+            if (maxX >= minX && maxY >= minY)
+            {
+                return new Box2(minX, minY, maxX, maxY);
+            }
+            return Box2.Zero;
+        }
+
+        public bool IntersectsWith(Box2 other)
+        {
+            return other._min.X < _max.X
+                && _min.X < other._max.X
+                && other._min.Y < _max.Y
+                && _min.Y < other._max.Y;
+        }
+
+        public bool TouchWith(Box2 other)
```

Comment:
I haven't thought too much about the `<=` or `<` when doing `Contains` and similar.
I think we should establish a consistent representation (write a documentation on the type on how it's expected to work) and make sure all functions follow that way of thinking.
Basically; should the edge of the box be contained or not?

Reply:
`IntersectsWith` means, they need a real intersection, where `TouchWith` means, it's also ok if they are directly next to each other (they are aligned on the edge).

We already have a `Contains(Vector2 point, bool boundaryInclusive)` and i'm not happy with the extra bool. The main problem here is naming. I think Contains should/can include the border, but what name for, if it should be NOT on the border?

Reply:
I still think that an important part of this is good documentation so that you can figure this out regardless of naming. For this type of situation I would probably take a look at the documentation to figure out what the alternatives are. But yeah we want good names for these... (maybe there is a good math terminology for this)
And if we get good naming we can also remove the current `Contains(Vector2 point, bool boundaryInclusive)`.